src/v2/guide/components-props.md

```diff
@@ -0,0 +1,314 @@
+
+HTML attribute names are case-insensitive, so browsers will interpret any uppercase characters as lowercase. That means when you're using in-DOM templates, camelCased prop names need to use their kebab-cased (hyphen-delimited) equivalents:
+
+``` js
+Vue.component('blog-post', {
+  // camelCase in JavaScript
+  props: ['postTitle'],
+  template: '<h3>{{ postTitle }}</h3>'
+})
+```
+
+``` html
+<!-- kebab-case in HTML -->
+<child post-title="hello!"></child>
+```
+
+Again, if you're using string templates, this limitation does not apply.
+
+## Static and Dynamic Props
+
+So far, you've seen props passed a static value, like in:
+
+```html
+<blog-post title="My journey with Vue"></blog-post>
+```
+
+You've also seen props assigned dynamically with `v-bind`, such as in:
+
+```html
+<blog-post v-bind:title="post.title"></blog-post>
+```
+
+In the two examples above, we happen to pass string values, but _any_ type of value can actually be passed to a prop.
+
+### Passing a Number
+
+```html
+<!-- Even though `42` is static, we need v-bind to tell Vue that -->
+<!-- this is a JavaScript expression rather than a string.       -->
+<blog-post v-bind:likes="42"></blog-post>
+
+<!-- Dynamically assign to the value of a variable. -->
+<blog-post v-bind:likes="post.likes"></blog-post>
+```
+
+### Passing a Boolean
+
+```html
+<!-- Including the prop with no value will imply `true`. -->
+<blog-post favorited></blog-post>
+
+<!-- Even though `false` is static, we need v-bind to tell Vue that -->
+<!-- this is a JavaScript expression rather than a string.          -->
+<base-input v-bind:favorited="false">
+
+<!-- Dynamically assign to the value of a variable. -->
+<base-input v-bind:favorited="post.currentUserFavorited">
+```
+
+### Passing an Array
+
+```html
+<!-- Even though the array is static, we need v-bind to tell Vue that -->
+<!-- this is a JavaScript expression rather than a string.            -->
+<blog-post v-bind:comment-ids="[234, 266, 273]"></blog-post>
+
+<!-- Dynamically assign to the value of a variable. -->
+<blog-post v-bind:comment-ids="post.commentIds"></blog-post>
+```
+
+### Passing an Object
+
+```html
+<!-- Even though the object is static, we need v-bind to tell Vue that -->
+<!-- this is a JavaScript expression rather than a string.             -->
+<blog-post v-bind:comments="{ id: 1, title: 'My Journey with Vue' }"></blog-post>
+
+<!-- Dynamically assign to the value of a variable. -->
+<blog-post v-bind:post="post"></blog-post>
+```
+
+### Passing the Properties of an Object
+
+If you want to pass all the properties of an object as props, you can use `v-bind` without an argument (`v-bind` instead of `v-bind:prop-name`). For example, given a `post` object:
+
+``` js
+post: {
+  id: 1,
+  title: 'My Journey with Vue'
+}
+```
+
+The following template:
+
+``` html
+<blog-post v-bind="post"></blog-post>
+```
+
+Will be equivalent to:
+
+``` html
+<blog-post
+  v-bind:id="post.id"
+  v-bind:title="post.title"
+></blog-post>
+```
```

Comment:
The syntax highlighting looks strange to me, here, is anyone else [seeing that](https://deploy-preview-1482--vuejs.netlify.com/v2/guide/components.html#Passing-Data-to-Child-Components-with-Props)?
<img width="229" alt="screen shot 2018-03-17 at 9 32 16 am" src="https://user-images.githubusercontent.com/2281088/37557175-4c9091ee-29c6-11e8-9f10-2f4fa8b43748.png">

Reply:
That's because I marked the syntax as JS instead of HTML. 😅 Fixed.